src/components/PostsSection.js

```diff
@@ -1,21 +1,89 @@
 import { useAuth } from "./../util/auth.js";
+import { useForm } from "react-hook-form";
+import FormField from "./FormField";
+import Section from "./Section";
+import SectionHeader from "./SectionHeader";
+import { ageGroups, eligibilityGroups, provinces, error, selectAll } from "./formConstants";
 
 function PostsSection(props) {
   const auth = useAuth();
   const router = useRouter();
-  
+  const { handleSubmit, register, errors } = useForm();
+
+  const [numPostals, setNumPostals] = useState(1);
+  const [pending, setPending] = useState(false);
+  const [groupError, setGroupError] = useState(false); 
+  const [areaError, setAreaError] = useState(false);
+
   let accountConfigured = auth.user.phone && auth.user.province && auth.user.postalcode;
 
+  const onSubmit = (data) => {
+    const allPostalCodes = document.querySelectorAll(".postalCodesInput");
+    const province = document.querySelector(".province");
+
+    if (document.querySelectorAll('input[type="checkbox"]:checked').length === 0) {
+      setGroupError(true);
+      const allCheckBoxes = document.querySelectorAll('input[type="checkbox"]');
+      for (let i = 0; i < allCheckBoxes.length; i++) {
+        allCheckBoxes[i].addEventListener("click", function() {
+          setGroupError(!this.checked)
+        });
+      }
+    }
+
+    data.postal = [];
+    allPostalCodes.forEach((postal) => {
+      if (!data.postal.includes(postal)) {
+        data.postal.push(postal.value.toUpperCase())
+      }
+    })
+
+    if (province.value && province.value != "--" && data.postal.length > 0 && data.postal[0] != "") {
+      setAreaError(true);
+
+      for (let i = 0; i < allPostalCodes.length; i++) {                                                                    
+        allPostalCodes[i].addEventListener("change", function(postal) {
+          if (!province.value && postal.value || province.value && !postal.value) {
+            setAreaError(false)
+          }
+        });
+      }
+
+      province.addEventListener("change", function(postal) {
+        if (!province.value && postal.value || province.value && !postal.value) {
+          setAreaError(false)
+        }
+      });
+
+    } 
+    else {
+      setGroupError(false);
+      setPending(true);
+
+      const selectedAgeGroups = [];
+      const selectedEligibilityGroups = [];
+  
+      const allSelectedGroups = document.querySelectorAll('input[type="checkbox"]:checked')
+      allSelectedGroups.forEach((group) => {
+        if (ageGroups.includes(group.id)) {
+          selectedAgeGroups.push(group.id);
+        } else if (eligibilityGroups.includes(group.id)) {
+          selectedEligibilityGroups.push(group.id);
+        }
+      })
+
+      data.selectedAgeGroups = selectedAgeGroups;
+      data.eligibilityGroups = selectedEligibilityGroups;  
+      data.province === "All" ? data.province = "CA" : data.province = "";
+
+      if (auth.user.admin) {
+        auth.postMessage(data);
```

Comment:
we should add a try/catch here and await the auth.postMessage() call. if it doesn't throw an error setPending(false) and show a confirmation message

Reply:
also, clear the form or show a success page?

Reply:
A message will be shown to the users when the Twilio side is connected.